lib/EventsGenerator.js

```diff
@@ -0,0 +1,174 @@
+   * @param {Array<string>} options.dotenvStub.vars
+   * @param {object} [options.dependencies]
+   * @param {object} [options.devDependencies]
+   * @memberof EventsGenerator
+   */
+  addEvents (eventDetails, tplActionPath, options = {}) {
+    // NOTE: it's important to load a fresh manifest now, as we do want to include the
+    // latest written data
+    if (eventDetails) {
+      const events = this.loadEventsManifest(this.configPath,
+        this.fullKeyToEventsManifest)
+      const { runtimeManifest, runtimePackageName } = this.loadRuntimeManifest(
+        this.configPath, this.fullKeyToManifest,
+        this.defaultRuntimePackageName)
+      if (!runtimeManifest.packages[runtimePackageName].actions[eventDetails.runtimeActionName]) {
+        this.addAction(eventDetails.runtimeActionName, tplActionPath, options)
+      }
+      this.setEventsManifestDetails(eventDetails, events, runtimePackageName)
+      this.writeEventsManifest(this.configPath, this.fullKeyToEventsManifest,
+        events)
+      const mapping = this.setDotEnvFileDetails(eventDetails)
+      this.writeDotEnvFile(mapping)
+    }
+  }
+
+  /** @private */
+  loadEventsManifest (configPath, fullKeyToEventsManifest) {
+    const config = utils.readYAMLConfig(this, configPath)
+    let events = fullKeyToEventsManifest.split('.').reduce((obj, k) => obj && obj[k], config) || {}
+    if (!events.registrations) {
+      events = {
+        registrations: {}
+      }
+    }
+    return events
+  }
+
+  /** @private */
+  writeEventsManifest (configPath, fullKeyToEventsManifest, events) {
+    utils.writeKeyYAMLConfig(this, configPath, fullKeyToEventsManifest, events)
+  }
+
+  /** @private */
+  writeDotEnvFile (valueMap) {
+    utils.appendVarsToDotenv(this, 'Provider metadata to provider id mapping', eventsProvidermetadataToProviderMapping, valueMap)
+  }
+
+  /** @private */
+  setEventsManifestDetails (regDetails, events, runtimePackageName) {
+    const eventsOfInterest = []
+    for (const providerMetadata of Object.keys(regDetails.selectedProvidersToEventMetadata)) {
+      eventsOfInterest.push({
+        provider_metadata: providerMetadata,
+        event_codes: regDetails.selectedProvidersToEventMetadata[providerMetadata].eventmetadata
+      })
+    }
+
+    events.registrations[regDetails.regName] = {
+      description: regDetails.regDesc,
+      events_of_interest: eventsOfInterest,
+      runtime_action: runtimePackageName + '/' + regDetails.runtimeActionName
+    }
+
+    return events
+  }
+
+  setDotEnvFileDetails (regDetails) {
+    const providerMetadataToProvidersExistingMap = getProviderMetadataToProvidersExistingMap(this.providerMetadataToProviderMappingDotEnv)
+    const newMapping = providerMetadataToProvidersExistingMap || {}
+    for (const providerMetadata of Object.keys(regDetails.selectedProvidersToEventMetadata)) {
+      // TODO: Ask for confirmation if provider id for an existing mapping changes
```

Comment:
are we leaving todos in here, was this overlooked? if not file an issue to take care of it later

Reply:
ill create an issue for this one

Reply:
https://github.com/adobe/generator-app-common-lib/issues/36